### PR TITLE
Offer a Leiningen plugin

### PR DESCRIPTION
**Pull request by vemv**
> Closes https://github.com/liquidz/antq/issues/79

First cut, let me know what you think!

I verified that it works locally:

image

(Clojure "1.11.99" is, of course, just a local hack)

### CODE REVIEW

File: src/antq/dep/leiningen.clj [hunk 1]
```diff
@@ -30,16 +44,14 @@
...
-                            :name  (if (u.dep/qualified-symbol?' dep-name)
-                                     (str dep-name)
-                                     (str dep-name "/" dep-name))
-                            :version version
+            :when (acceptable-version? version)]
+        (r/map->Dependency {:project      :leiningen
+                            :type         :java
+                            :file         file-path
+                            :name         (normalize-name dep-name)
+                            :version      version
```
Comment (liquidz) [inline src/antq/dep/leiningen.clj:54]: I'm not using alignment in this code base.
```suggestion
        (r/map->Dependency {:project :leiningen
                            :type :java
                            :file file-path
                            :name (normalize-name dep-name)
                            :version version
```

File: src/leiningen/antq.clj [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                      (into managed-dependencies)
+                      (distinct)
+                      (keep (fn [[dep-name version]]
+                              (when (dep.lein/acceptable-version? version)
+                                (r/map->Dependency {:project      :leiningen
+                                                    :type         :java
+                                                    :file         "project.clj"
+                                                    :name         (dep.lein/normalize-name dep-name)
+                                                    :version      version
+                                                    :repositories repos}))))
```
Comment (liquidz) [inline src/leiningen/antq.clj:34]: Same here.
```suggestion
                                (r/map->Dependency {:project :leiningen
                                                    :type :java
                                                    :file "project.clj"
                                                    :name (dep.lein/normalize-name dep-name)
                                                    :version version
                                                    :repositories repos}))))
```

**Review by liquidz — CHANGES_REQUESTED** [review]
@vemv Thanks! Overall the PR looks good and it works great!

Could you update CHANGELOG.adoc too?

**Comment by vemv** [comment]
Thanks for the review! Ready again. Build is green in https://github.com/reducecombine/antq/actions

**Review by liquidz — APPROVED** [review]
@vemv Thanks! LGTM

**Comment by vemv** [comment]
Cheers 🙌

I was thinking of adding an end-to-end test for the Lein plugin. Sounds good?

The way I did it for another lein plugin was with a bash script. This way it's natural to express various steps e.g. `lein install`, then running the plugin and expecting non-zero, then running the plugin and expecting zero, etc.

Example: https://github.com/rm-hull/lein-nvd/blob/ed8780007256a3c41dfad144ababbd4b05727995/.github/integration_test.sh

**Comment by liquidz** [comment]
@vemv Awesome! It's good to add tests!

**Comment by vemv** [comment]
Thanks for the confirmation! Will do

**Pull request merged**